### PR TITLE
enable credentials for pushing manylinux images

### PR DESCRIPTION
**Pull request by janeyx99**
Adds a mechanism so that the build jobs will push the images when the PR is merged to master.

Uses secrets DOCKER_ID and DOCKER_TOKEN for Docker Hub.

### CODE REVIEW

File: .github/workflows/build-manywheel-images.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 name: Build manywheel docker images
 
 on:
+  push:
```
Comment (samestep) [inline .github/workflows/build-manywheel-images.yml:4]: do you need to specify `paths` here too?
```suggestion
  push:
    paths:
      - manywheel/Dockerfile
```

File: manywheel/build_docker.sh [hunk 2]
```diff
@@ -57,9 +57,10 @@ DOCKER_IMAGE=${DOCKER_REGISTRY}/pytorch/manylinux-builder:${DOCKER_TAG}
     docker tag ${DOCKER_IMAGE} ${LEGACY_DOCKER_IMAGE}
 )
 
-if [[ -n ${WITH_PUSH} ]]; then
+if [[ "${WITH_PUSH}" == true ]]; then
     (
         set -x
+        echo "${DOCKER_TOKEN}" | docker login -u "${DOCKER_ID}" --password--stdin
```
Comment (seemethere) [inline manywheel/build_docker.sh:63]: Let's try and have the login just be a separate step in the workflow, we don't want to have users to have to provide these environment variables in order to actually run this script

File: manywheel/build_docker.sh [hunk 3]
```diff
@@ -57,9 +57,10 @@ DOCKER_IMAGE=${DOCKER_REGISTRY}/pytorch/manylinux-builder:${DOCKER_TAG}
     docker tag ${DOCKER_IMAGE} ${LEGACY_DOCKER_IMAGE}
 )
 
-if [[ -n ${WITH_PUSH} ]]; then
+if [[ "${WITH_PUSH}" == true ]]; then
```
Comment (seemethere) [inline manywheel/build_docker.sh:60]: What's the reasoning behind this change? Wouldn't it also just work without needing to change the actual equals value here?

File: .github/workflows/build-manywheel-images.yml (hunk 1 above)
Comment (janeyx99) [reply .github/workflows/build-manywheel-images.yml:4]: oh good call

File: manywheel/build_docker.sh (hunk 3 above)
Comment (janeyx99) [reply manywheel/build_docker.sh:60]: It's cuz I set it to ${{ github.event_name == 'push' }}, which I think would be 'false' if we're not pushing, which would pass the `-n`, no?

Comment (seemethere) [reply manywheel/build_docker.sh:60]: Oh that's true, we should probably document this in the README.md

Perhaps we can also make this change for the rest of the docker build scripts as well?

**Review by seemethere — APPROVED** [review]

**Pull request merged**